File: app/src/main/java/com/artamonov/millionplanets/modules/ModulesListActivity.java
package com.artamonov.millionplanets.modules;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import androidx.appcompat.app.AppCompatActivity;
import com.artamonov.millionplanets.R;

public class ModulesListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.modules_list);
    }

    public void onWeapons(View view) {
        Intent intent = new Intent(this, ModulesActivity.class);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            startActivity(intent, ActivityOptions.makeSceneTransitionAnimation(this).toBundle());
        } else {
            startActivity(intent);
        }
    }
}

File: app/src/main/java/com/artamonov/millionplanets/modules/ModulesListActivity.kt
package com.artamonov.millionplanets.modules

import android.app.ActivityOptions
import android.content.Intent
import android.os.Build
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.artamonov.millionplanets.R

class ModulesListActivity : AppCompatActivity(R.layout.modules_list) {

    fun onWeapons(view: View?) {
        val intent = Intent(this, ModulesActivity::class.java)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            startActivity(intent, ActivityOptions.makeSceneTransitionAnimation(this).toBundle())
        } else {
            startActivity(intent)
        }
    }
}